### PR TITLE
permission-node: rename and adjust policy return type to reduce nesting

### PR DESCRIPTION
**Pull request by mtlewis**
## Hey, I just made a Pull Request!

Addresses some awkward naming & nesting in the policy return type by flattening the type and doing some renaming.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by mtlewis** [comment]
See also the impact that this will have on the PRFC branch in https://github.com/backstage/backstage/commit/e526ffc692fb3fe47a39ef222df13b283c288a9d.

**Review by Joonpark13 — APPROVED** [review]

**Review by timbonicus — APPROVED** [review]

File: plugins/permission-node/src/integration/createConditionExports.test.ts [hunk 1]
```diff
@@ -63,11 +64,14 @@ describe('createConditionExports', () => {
 
   describe('createConditions', () => {
```
Comment (joeporpeglia) [inline plugins/permission-node/src/integration/createConditionExports.test.ts:65]:
```suggestion
  describe('createPolicyDecisions', () => {
```

**Review by joeporpeglia — APPROVED** [review]
Looks great!

File: plugins/permission-node/src/integration/createConditionExports.test.ts (hunk 1 above)
Comment (mtlewis) [reply plugins/permission-node/src/integration/createConditionExports.test.ts]: Thanks!

**Review by Rugvip — APPROVED** [review]
👌 - loving the "decision" naming x)

**Pull request merged**